mne_bids/path.py

```diff
@@ -56,44 +55,80 @@ class BIDSPath(object):
     space : str | None
         The coordinate space for an anatomical file. Corresponds to "space".
     split : int | None
-        The split of the continuous recording file for ``.fif`` data. Corresponds to "split".
-    prefix : str | None
-        The prefix for the filename to be created. E.g., a path to the folder
-        in which you wish to create a file with this name.
-    kind : str | None
-        The filename kind. This is the entity after the last ``_`` before the extension. 
-        E.g., ``'ieeg'``.
+        The split of the continuous recording file for ``.fif`` data.
+        Corresponds to "split".
+    suffix : str | None
+        The suffix of the filename. Will be appended to the filename
+        before the extension (``*_<suffix>.<extension>``).
+        The following filename suffix's are accepted:
+        'meg', 'markers', 'eeg', 'ieeg', 'T1w',
+        'participants', 'scans', 'electrodes', 'coordsystem',
+        'channels', 'events', 'headshape', 'digitizer',
+        'behav', 'phsyio', 'stim'
     extension : str | None
-        The extension of the filename. E.g., ``'.json'``.
+        The extension for the filename. E.g., '.edf'
+    kind : str | None
+        The suffix for the path, which refers to the directory name after
+        ``sub-<label>/[ses-<label>/]`` in the BIDS specification. For example,
+        this can be ``meg``, ``eeg``, ``ieeg``, or ``anat``.
+    bids_root : str | None
+        The bids_root directory of the BIDS dataset.
+
+    Attributes
+    ----------
+    entities : dict
+        The dictionary of the BIDS entities and their values:
+        ``subject``, ``session``, ``task``, ``acquisition``,
+        ``run``, ``processing``, ``space``, ``recording`` and ``suffix``.
+    basename : str
+        The basename of the file path. Similar to `os.path.basename(fpath)`.
+    fpath : str
+        The full file path.
 
     Examples
     --------
-    >>> bids_basename = make_bids_basename(subject='test', session='two', 
-                                           task='mytask', kind='ieeg', extension='.edf')
-    >>> print(bids_basename)
+    >>> bids_path = make_bids_basename(subject='test', session='two',
+                                           task='mytask', suffix='ieeg',
+                                           extension='.edf'))
+    >>> print(bids_path)
     sub-test_ses-two_task-mytask_ieeg.edf
-    >>> bids_basename
+    >>> bids_path
     BIDSPath(sub-test_ses-two_task-mytask_ieeg.edf)
     >>> # copy and update multiple entities at once
-    >>> new_basename = bids_basename.copy().update(subject='test2', session='one')
+    >>> new_basename = bids_path.copy().update(subject='test2',
+                                                   session='one')
     >>> print(new_basename)
     sub-test2_ses-one_task-mytask_ieeg.edf
```

Comment:
to me this should be:

```suggestion
    sub-test2/ses-one/ieeg/sub-test2_ses-two_task-mytask_ieeg.edf
```
ie without bids_root but still full path inside the bids dataset. If you want
just `sub-test2_ses-two_task-mytask_ieeg.edf` you should get it with:
`bids_path .basename`